File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/wrapping_and_braces/JavaExtendsImplementsListSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.wrapping_and_braces;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavaExtendsImplementsListSettings.class)
public interface JavaExtendsImplementsListSettings {

  Optional<Wrapping> wrapping();

  Optional<Boolean> alignWhenMultiline();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/wrapping_and_braces/JavaExtendsImplementsListSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.wrapping_and_braces

import com.fasterxml.jackson.annotation.JsonProperty

data class JavaExtendsImplementsListSettings(

    @JsonProperty("Wrapping")
    val wrapping: Wrapping? = null,

    @JsonProperty("Align when multiline")
    val alignWhenMultiline: Boolean? = null
)
